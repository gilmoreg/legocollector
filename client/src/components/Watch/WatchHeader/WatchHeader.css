.WatchHeader {
    max-width: 760px;
    border: 1px solid black;
    border-bottom: none;
    border-radius: 5px 5px 0px 0px;

    display: flex;
    justify-content: space-around;
}

.WatchHeader .thumbnail {
    flex: 0 0 20%;
    align-self: center;
    border: 1px solid red;
    padding: 5px;
}

.WatchHeader .title {
    flex: 0 0 50%;
    align-self: center;
    text-align: left;
    border: 1px solid red;
}

.WatchHeader .stock {
    flex: 0 0 20%;
    color: green;
    align-self: center;
    font-size: 36px;
    border: 1px solid red;
}

.WatchHeader img {
    width: 100%;
}