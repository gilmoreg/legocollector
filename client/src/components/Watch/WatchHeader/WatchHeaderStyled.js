import styled from 'styled-components';

export const StyledWatchHeader = styled.div`
  max-width: 760px;
  background-color: rgba(255,246,229,0.7);
  border-bottom: none;
  border-radius: 5px 5px 0px 0px;
  margin: 0 auto;
  display: flex;
  justify-content: space-around;
  z-index: -2;
`;

export const StyledThumbnail = styled.div`
  flex: 0 0 20%;
  align-self: center;
  padding: 5px;
  padding-top: 10px;
`;

export const StyledTitle = styled.div`
  flex: 0 0 50%;
  align-self: center;
  text-align: left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  @media (min-width: 640px) {
    font-size: 24px;
  }
`;

export const StyledStock = styled.div`
  flex: 0 0 20%;
  color: green;
  align-self: center;
  font-size: 36px;
  padding-top: 5px;

  @media (min-width: 640px) {
    font-size: 60px;
  }
`;

export const StyledStockP = styled.p`
  margin-top: 5px;
  margin-bottom: 5px;
  font-size: 14px;
  color: black;

  @media (min-width: 640px) {
    font-size: 24px;
  }
`;

export const StyledImg = styled.img`
  width: 100%;
`;
