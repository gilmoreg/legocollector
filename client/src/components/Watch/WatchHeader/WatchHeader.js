import React from 'react';
import PropTypes from 'prop-types';

import './WatchHeader.css';

const WatchHeader = props => (
  <div className="WatchHeader">
    <div className="thumbnail">
      <a href={props.watch.url} target="_blank" rel="noopener noreferrer" title={props.watch.title}>
        <img src={props.watch.image} alt={props.watch.title} />
      </a>
    </div>
    <div className="title">
      <a href={props.watch.url} target="_blank" rel="noopener noreferrer" title={props.watch.title}>
        {props.watch.title}
      </a>
    </div>
    <div className="stock">
      { props.watch.currentStock > 999 ?
        '999+'
        :
        props.watch.currentStock
      }
    </div>
  </div>
);

WatchHeader.propTypes = {
  watch: PropTypes.shape({
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    currentStock: PropTypes.number.isRequired,
  }).isRequired,
};

export default WatchHeader;
