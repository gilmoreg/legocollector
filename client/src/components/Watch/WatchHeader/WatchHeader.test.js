import React from 'react';
import { shallow } from 'enzyme';
import WatchHeader from './WatchHeader';

it('renders without crashing', () => {
  const watch = {
    title: 'test',
    image: 'test',
    url: 'test',
    currentStock: 0,
  }
  const wrapper = shallow(
    <WatchHeader watch={watch} />
  );
  expect(wrapper).toMatchSnapshot();
});
