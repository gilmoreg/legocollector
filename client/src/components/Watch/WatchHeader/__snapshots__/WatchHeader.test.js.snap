// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WatchHeader
    watch={
      Object {
        "image": "https://images-na.ssl-images-amazon.com/images/I/413yqVUgjcL._SL160_.jpg",
        "stock_levels": Array [
          Object {
            "datetime": "test",
            "stock_level": 1,
          },
          Object {
            "datetime": "test",
            "stock_level": 1,
          },
        ],
        "title": "LEGO Ideas NASA Apollo Saturn V 21309 Building Kit",
        "url": "https://www.amazon.com/LEGO-Ideas-Apollo-Saturn-Building/dp/B01MUANC80?psc=1&SubscriptionId=AKIAJ57KMBTCZPRJ5I5Q&tag=gilmoreg-20&linkCode=xm2&camp=2025&creative=165953&creativeASIN=B01MUANC80",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="thumbnail"
        >
          <a
            href="https://www.amazon.com/LEGO-Ideas-Apollo-Saturn-Building/dp/B01MUANC80?psc=1&SubscriptionId=AKIAJ57KMBTCZPRJ5I5Q&tag=gilmoreg-20&linkCode=xm2&camp=2025&creative=165953&creativeASIN=B01MUANC80"
            rel="noopener noreferrer"
            target="_blank"
            title="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
          >
            <img
              alt="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
              src="https://images-na.ssl-images-amazon.com/images/I/413yqVUgjcL._SL160_.jpg"
            />
          </a>
        </div>,
        <div
          className="title"
        >
          <a
            href="https://www.amazon.com/LEGO-Ideas-Apollo-Saturn-Building/dp/B01MUANC80?psc=1&SubscriptionId=AKIAJ57KMBTCZPRJ5I5Q&tag=gilmoreg-20&linkCode=xm2&camp=2025&creative=165953&creativeASIN=B01MUANC80"
            rel="noopener noreferrer"
            target="_blank"
            title="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
          >
            LEGO Ideas NASA Apollo Saturn V 21309 Building Kit
          </a>
        </div>,
        <div
          className="stock"
        >
          1
        </div>,
      ],
      "className": "WatchHeader",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="https://www.amazon.com/LEGO-Ideas-Apollo-Saturn-Building/dp/B01MUANC80?psc=1&SubscriptionId=AKIAJ57KMBTCZPRJ5I5Q&tag=gilmoreg-20&linkCode=xm2&camp=2025&creative=165953&creativeASIN=B01MUANC80"
            rel="noopener noreferrer"
            target="_blank"
            title="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
          >
            <img
              alt="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
              src="https://images-na.ssl-images-amazon.com/images/I/413yqVUgjcL._SL160_.jpg"
            />
          </a>,
          "className": "thumbnail",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
              src="https://images-na.ssl-images-amazon.com/images/I/413yqVUgjcL._SL160_.jpg"
            />,
            "href": "https://www.amazon.com/LEGO-Ideas-Apollo-Saturn-Building/dp/B01MUANC80?psc=1&SubscriptionId=AKIAJ57KMBTCZPRJ5I5Q&tag=gilmoreg-20&linkCode=xm2&camp=2025&creative=165953&creativeASIN=B01MUANC80",
            "rel": "noopener noreferrer",
            "target": "_blank",
            "title": "LEGO Ideas NASA Apollo Saturn V 21309 Building Kit",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "alt": "LEGO Ideas NASA Apollo Saturn V 21309 Building Kit",
              "src": "https://images-na.ssl-images-amazon.com/images/I/413yqVUgjcL._SL160_.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="https://www.amazon.com/LEGO-Ideas-Apollo-Saturn-Building/dp/B01MUANC80?psc=1&SubscriptionId=AKIAJ57KMBTCZPRJ5I5Q&tag=gilmoreg-20&linkCode=xm2&camp=2025&creative=165953&creativeASIN=B01MUANC80"
            rel="noopener noreferrer"
            target="_blank"
            title="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
          >
            LEGO Ideas NASA Apollo Saturn V 21309 Building Kit
          </a>,
          "className": "title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "LEGO Ideas NASA Apollo Saturn V 21309 Building Kit",
            "href": "https://www.amazon.com/LEGO-Ideas-Apollo-Saturn-Building/dp/B01MUANC80?psc=1&SubscriptionId=AKIAJ57KMBTCZPRJ5I5Q&tag=gilmoreg-20&linkCode=xm2&camp=2025&creative=165953&creativeASIN=B01MUANC80",
            "rel": "noopener noreferrer",
            "target": "_blank",
            "title": "LEGO Ideas NASA Apollo Saturn V 21309 Building Kit",
          },
          "ref": null,
          "rendered": "LEGO Ideas NASA Apollo Saturn V 21309 Building Kit",
          "type": "a",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "1",
          "className": "stock",
        },
        "ref": null,
        "rendered": "1",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="thumbnail"
          >
            <a
              href="https://www.amazon.com/LEGO-Ideas-Apollo-Saturn-Building/dp/B01MUANC80?psc=1&SubscriptionId=AKIAJ57KMBTCZPRJ5I5Q&tag=gilmoreg-20&linkCode=xm2&camp=2025&creative=165953&creativeASIN=B01MUANC80"
              rel="noopener noreferrer"
              target="_blank"
              title="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
            >
              <img
                alt="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
                src="https://images-na.ssl-images-amazon.com/images/I/413yqVUgjcL._SL160_.jpg"
              />
            </a>
          </div>,
          <div
            className="title"
          >
            <a
              href="https://www.amazon.com/LEGO-Ideas-Apollo-Saturn-Building/dp/B01MUANC80?psc=1&SubscriptionId=AKIAJ57KMBTCZPRJ5I5Q&tag=gilmoreg-20&linkCode=xm2&camp=2025&creative=165953&creativeASIN=B01MUANC80"
              rel="noopener noreferrer"
              target="_blank"
              title="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
            >
              LEGO Ideas NASA Apollo Saturn V 21309 Building Kit
            </a>
          </div>,
          <div
            className="stock"
          >
            1
          </div>,
        ],
        "className": "WatchHeader",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="https://www.amazon.com/LEGO-Ideas-Apollo-Saturn-Building/dp/B01MUANC80?psc=1&SubscriptionId=AKIAJ57KMBTCZPRJ5I5Q&tag=gilmoreg-20&linkCode=xm2&camp=2025&creative=165953&creativeASIN=B01MUANC80"
              rel="noopener noreferrer"
              target="_blank"
              title="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
            >
              <img
                alt="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
                src="https://images-na.ssl-images-amazon.com/images/I/413yqVUgjcL._SL160_.jpg"
              />
            </a>,
            "className": "thumbnail",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
                src="https://images-na.ssl-images-amazon.com/images/I/413yqVUgjcL._SL160_.jpg"
              />,
              "href": "https://www.amazon.com/LEGO-Ideas-Apollo-Saturn-Building/dp/B01MUANC80?psc=1&SubscriptionId=AKIAJ57KMBTCZPRJ5I5Q&tag=gilmoreg-20&linkCode=xm2&camp=2025&creative=165953&creativeASIN=B01MUANC80",
              "rel": "noopener noreferrer",
              "target": "_blank",
              "title": "LEGO Ideas NASA Apollo Saturn V 21309 Building Kit",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "alt": "LEGO Ideas NASA Apollo Saturn V 21309 Building Kit",
                "src": "https://images-na.ssl-images-amazon.com/images/I/413yqVUgjcL._SL160_.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="https://www.amazon.com/LEGO-Ideas-Apollo-Saturn-Building/dp/B01MUANC80?psc=1&SubscriptionId=AKIAJ57KMBTCZPRJ5I5Q&tag=gilmoreg-20&linkCode=xm2&camp=2025&creative=165953&creativeASIN=B01MUANC80"
              rel="noopener noreferrer"
              target="_blank"
              title="LEGO Ideas NASA Apollo Saturn V 21309 Building Kit"
            >
              LEGO Ideas NASA Apollo Saturn V 21309 Building Kit
            </a>,
            "className": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "LEGO Ideas NASA Apollo Saturn V 21309 Building Kit",
              "href": "https://www.amazon.com/LEGO-Ideas-Apollo-Saturn-Building/dp/B01MUANC80?psc=1&SubscriptionId=AKIAJ57KMBTCZPRJ5I5Q&tag=gilmoreg-20&linkCode=xm2&camp=2025&creative=165953&creativeASIN=B01MUANC80",
              "rel": "noopener noreferrer",
              "target": "_blank",
              "title": "LEGO Ideas NASA Apollo Saturn V 21309 Building Kit",
            },
            "ref": null,
            "rendered": "LEGO Ideas NASA Apollo Saturn V 21309 Building Kit",
            "type": "a",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "1",
            "className": "stock",
          },
          "ref": null,
          "rendered": "1",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
